src/main/java/server/Server.java

```diff
@@ -367,9 +367,19 @@ public void run() {
       }
       InputStreamReader isr = new InputStreamReader(is);
       BufferedReader br = new BufferedReader(isr);
+      FileWriter file;
       while (true) {
         try {
           telemetryData = new JSONObject(br.readLine());
+          try {
+            file = new FileWriter("TelemetryData.txt");
```

Comment:
Try moving the file creation outside the loop so we only have one file.

Reply:
Done.